packages/core/src/wallet/metamaskBridge.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  DisconnectedStarknetWindowObject,
+  RpcMessage,
+} from "../StarknetWindowObject"
+import wallets, { WalletProvider } from "../discovery"
+import type { MetaMaskProvider } from "@consensys/get-starknet/dist/type"
+import detectEthereumProvider from "@metamask/detect-provider"
+import type { ProviderInterface } from "starknet"
+
+class EmptyMetaMaskProvider implements DisconnectedStarknetWindowObject {
+  id: string
+  name: string
+  icon: string
+  version = "0.0.0"
+  isConnected = false as const
+  provider: undefined | ProviderInterface
+  constructor(
+    private metamaskProvider: MetaMaskProvider,
+    walletInfo: WalletProvider,
+  ) {
+    this.id = walletInfo.id
+    this.name = walletInfo.name
+    this.icon = walletInfo.icon
+  }
+  request<T extends RpcMessage>(): Promise<T["result"]> {
+    throw new Error("Wallet not enabled")
+  }
+  async enable(options?: { starknetVersion?: "v4" | "v5" }): Promise<string[]> {
+    const { MetaMaskSnapWallet } = await import("@consensys/get-starknet")
+    const metaMaskSnapWallet = new MetaMaskSnapWallet(this.metamaskProvider)
+
+    Object.assign(this, metaMaskSnapWallet)
+    this.constructor.prototype = metaMaskSnapWallet.constructor.prototype
```

Comment:
constructor is read only, are you looking for inherit the props from constructor or also including the private & public methods of the class?

Reply:
@stanleyyconsensys 
I'd like to replace "this" with the metaMaskSnapWallet instance, so that all metaMaskSnapWallet properties including methods and implementations will override implementations on "this".